### PR TITLE
feat: remove podman CLI impl copy and use podman ext API

### LINKED ISSUE #44: Remove copy of podman CLI implementation and migrate to new API when implemented
Current implementation is copied form podman extension and should be removed when podman desktop has new API to call podman commands https://github.com/containers/podman-desktop/issues/5990.

### PR DESCRIPTION
**Pull request by dgolovin**
Fix #44.

Requires https://github.com/containers/podman-desktop/issues/5990 to be merged and podman API package published.

### CODE REVIEW

**Review by jeffmaury — COMMENTED** [review]
What is the status of this PR ?

**Comment by dgolovin** [comment]
> What is the status of this PR ?

Depends on https://github.com/containers/podman-desktop/pull/9061.

**Review by axel7083 — COMMENTED** [review]
If we start depending on high version of podman-desktop we need to enforce it through 

https://github.com/redhat-developer/podman-desktop-redhat-account-ext/blob/b3f17086b61208e14285b4086a99413f31372f14/Containerfile#L23

and

https://github.com/redhat-developer/podman-desktop-redhat-account-ext/blob/b3f17086b61208e14285b4086a99413f31372f14/package.json#L12

so user cannot use it on version without this feature

**Comment by dgolovin** [comment]
> If we start depending on high version of podman-desktop we need to enforce it through
> 
> https://github.com/redhat-developer/podman-desktop-redhat-account-ext/blob/b3f17086b61208e14285b4086a99413f31372f14/Containerfile#L23
> 
> and
> 
> https://github.com/redhat-developer/podman-desktop-redhat-account-ext/blob/b3f17086b61208e14285b4086a99413f31372f14/package.json#L12
> 
> so user cannot use it on version without this feature

I can do unit test for that, but there should be probably some sort of extension validation tool or unit tests suit.

**Review by jeffmaury — APPROVED** [review]

File: src/podman-cli.ts [hunk 1]
```diff
@@ -16,12 +16,10 @@
  * SPDX-License-Identifier: Apache-2.0
  ***********************************************************************/
 import type { RunError, RunResult } from '@podman-desktop/api';
-import { configuration, process as podmanProcess, provider } from '@podman-desktop/api';
+import * as pdApi from '@podman-desktop/api';
```
Comment (dgolovin) [inline src/podman-cli.ts:19]:
```suggestion
import * as podmanDesktopAPI from '@podman-desktop/api';
```

File: src/podman-cli.ts [hunk 2]
```diff
@@ -45,51 +43,27 @@ export const PODMAN_COMMANDS = {
...
+const podmanApiDummy = {
+  exec: (): Promise<RunResult> => {
+    throw Error('Podman extension API is not available.');
+  },
+};
 
-export interface InstalledPodman {
-  version: string;
+function getPodmanApi(): PodmanExtensionApi {
+  const podmanExports = pdApi.extensions.getExtension<PodmanExtensionApi>('podman-desktop.podman')?.exports;
```
Comment (dgolovin) [inline src/podman-cli.ts:53]:
```suggestion
  const podmanExports = podmanDesktopAPI.extensions.getExtension<PodmanExtensionApi>('podman-desktop.podman')?.exports;
```

File: src/podman-cli.ts [hunk 3]
```diff
@@ -193,7 +167,7 @@ export async function runStartPodmanMachine(machineName): Promise<RunResult> {
 }
 
 export function getRunningPodmanMachineName(): string | undefined {
-  const conns = provider.getContainerConnections();
+  const conns = pdApi.provider.getContainerConnections();
```
Comment (dgolovin) [inline src/podman-cli.ts:170]:
```suggestion
  const conns = podmanDesktopAPI.provider.getContainerConnections();
```

**Pull request merged**